docs/release-cycle.md

```diff
@@ -0,0 +1,81 @@
+   - Coordinate with the stable branch manager.
+ - Plan/prioritize features, testing work and bugs
+   - This is usually done in the SIG Cluster Lifecycle meetings, and a dedicated v1.X planning doc is made and referenced from the [meeting notes](https://docs.google.com/document/d/1deJYPIF4LmhGjDVaqrswErIrV7mtwJgovtLnPCDxP7U/edit#).
+   - There people sign up for working on the high level features and objectives for the release
+   - Create the right feature issues in `kubernetes/features` for what you want to accomplish for the quarter. Deadline is often the last day of the first month.
+ - Go through _every_ open issue in the kubeadm repo.
+   - As there are a small number of core maintainers working on kubeadm, there often isn't time to answer all feature requests or bugfixes, or to close old, outdated ones.
+   - As a general practice, every issue must be triaged **at least** once a quarter.
+ - Write proposals for new and larger features
+   - Often starting as a Google Doc that is easy to comment on/let others collaborate on.
+   - Get consensus on the design in the SIG
+   - Convert to markdown, send PR to `kubernetes/community` and loop in the rest of the community (other relevant SIGs, if not looped in already)
+   - Update as needed by review comments, get LGTM and merge.
+ - Start coding new features, fix bugs and increase unit testing coverage levels
+ - Ensure that the [`kubeadm-gce`](https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#kubeadm-gce) is master blocking.
+   - "Master blocking" means that this is the dashboard the release team looks at when they decide whether to cut an alpha.X release or not.
+ - Cleanup work. As kubeadm supports deploying clusters one minor version below the CLI version, support for v1.7-specific workarounds may be dropped when the v1.9 cycle starts
+   - A good example of this is Bootstrap Tokens and RBAC rules.
+   - In v1.7, alpha Bootstrap Tokens were used. On upgrade to v1.8 they were slightly modified to use beta features and resource names
+   - In v1.7, the `extensions/v1beta1` and `rbac.authorization.k8s.io/v1beta1` groups were used. In v1.8, `apps/v1beta2` and `rbac.authorization.k8s.io/v1` were added. Early in the v1.9 we can switch over to use `apps/v1beta2` and `rbac.authorization.k8s.io/v1` as those groups will always exist for the supported versions.
+   - These are good examples of cleanup work that can be done. Also, `cmd/kubeadm/app/constants.MinimumControlPlaneVersion` should be bumped (in this example from `v1.7.0` to `v1.8.0`)
+   - You might want to wait some time from the previous .0 release to make it easier to cherrypick possible bug fixes.
+
+#### Second month of the quarter (coding)
+
+ - Code, code, code :)
+ - Create new e2e tests to cover functionality of kubeadm or that kubeadm depends on
+   - Example: In the v1.8 cycle, e2e tests were added for the Node Authorizer and Bootstrap Tokens, both of them features kubeadm depend heavily on
```

Comment:
It should be really useful to refer documentation about how kubeadm e2e test work (the one we were sharing with @pipejakob some times ago)